tiledb/tests/test_multiprocessing.py

```diff
@@ -24,10 +27,80 @@ def test_warn_on_fork():
     # accidentally fork(): multiprocessing will default to "spawn" on
     # Linux.
 
+    def wrap_fork():
+        os_fork = os.fork
+
+        @wraps(os_fork)
+        def wrapper():
+            warnings.warn(
+                "TileDB is a multithreading library and deadlocks are "
+                "likely if fork() is called after a TileDB array has "
+                "been created or accessed. "
+                "To safely use TileDB with multiprocessing or "
+                "concurrent.futures, choose 'spawn' as the start "
+                "method for child processes.",
+                UserWarning,
+            )
+            return os_fork()
+
+        return wrapper
+
+    import tiledb  # noqa
+
+    if sys.version_info < (3, 12):
+        os.fork = wrap_fork()
```

Comment:
These lines would be moved into the tiledb package if we agree to go this route.